src/cfi.rs

```diff
@@ -639,11 +641,15 @@ struct CfiEntryCommon<R: Reader> {
 /// end-of-entries sentinel, return `Ok(None)`. Otherwise, return
 /// `Ok(Some(tuple))`, where `tuple.0` is the start of the next entry and
 /// `tuple.1` is the parsed CFI entry data.
-fn parse_cfi_entry_common<Section, R>(input: &mut R) -> Result<Option<CfiEntryCommon<R>>>
+fn parse_cfi_entry_common<Section, R>(
+    section: Section,
```

Comment:
By passing `Section`, we know its pointing to the start of the whole section, whereas with an `&R`, it could be at some offset within the section.